ADAL/src/cache/ios/ADKeychainTokenCache.m

```diff
@@ -451,6 +451,34 @@ - (BOOL)removeAllForUserId:(NSString * __nonnull)userId
     return deleteSuccessful;
 }
 
+- (BOOL)removeAllForUserId:(NSString *)userId error:(ADAuthenticationError *__autoreleasing  _Nullable *)error
+{
+    AD_LOG_WARN_DICT(([NSString stringWithFormat:@"Removing all items for user <%@>", userId]), nil,
+                     (@{ @"operation" : @"removeAllForClientId:", @"userId" : userId }), nil);
+
+    BOOL deleteSuccessful = YES;
+    NSArray *items = [self allItems:nil];
+
+    ADAuthenticationError *adError = nil;
+
+    for (ADTokenCacheItem *item in items)
+    {
+        if ([userId isEqualToString:[[item userInformation] userId]])
+        {
+            [self removeItem:item error:&adError];
+            if (adError)
+            {
+                deleteSuccessful = NO;
+                if (error)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if|for|do|while|switch statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS134)